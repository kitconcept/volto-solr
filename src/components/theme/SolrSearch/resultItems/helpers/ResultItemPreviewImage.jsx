import { Link } from 'react-router-dom';
import config from '@plone/volto/registry';
import LegacyImage from './LegacyImage';

export const previewImageContent = ({
  '@id': id,
  extras: { image_scales: rawImageScales },
  image_field,
}) => {
  let image_scales;
  try {
    image_scales = JSON.parse(rawImageScales);
  } catch {}
  // In case of json error, image_scales will be undefined..
  return {
    '@id': id,
    image_scales,
    image_field,
  };
};

const LinkToImage = ({ item, children }) => (
  <Link to={item['@id']}>{children}</Link>
);

const EmptyWrapper = ({ item, children }) => <>{children}</>;

const ResultItemPreviewImage = ({ item, Wrapper = LinkToImage, ...rest }) => {
  const content = previewImageContent(item);
  Wrapper = Wrapper !== null ? Wrapper : EmptyWrapper;
  if (content.image_scales) {
    // Show the wrapper also conditionally.
    let Image;
    if (!config.settings.contentTypeSearchResultAlwaysUseLegacyImage) {
      Image = config.getComponent({ name: 'Image' }).component;
    }
    if (!Image) {
      // Volto < 17 does not have the Image component. We provide a legacy fallback.
      // This will be a simple sourced image, no srcset, with the required resolution
      // to be used.
      if (rest.sizes) {
        throw new Error(
          'The `sizes` attribute of ResultItemPreviewImage does not work with the legacy image component. You must use Volto >= 17 with the Volto Image component for this to work.',
        );
      }
      return (
        <Wrapper item={item}>
          <LegacyImage
            item={content}
            /* Default hints provided, can be overridden via rest */
            className="previewImage"
            loading="lazy"
            width="250"
            height="125"
            {...rest}
          />
        </Wrapper>
      );
    }
    return (
      <Wrapper item={item}>
        <Image
          item={content}
          alt={item.title}
          /* Default hints provided, can be overridden via rest */
          className="previewImage"
          loading="lazy"
          sizes="250px"
          /* width + height practically mandatory, see https://github.com/plone/volto/issues/5096 */
          width="250"
          height="125"
          {...rest}
        />
      </Wrapper>
    );
  } else {
    return null;
  }
};

export default ResultItemPreviewImage;
