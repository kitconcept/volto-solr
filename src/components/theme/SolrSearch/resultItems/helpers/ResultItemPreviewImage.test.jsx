import { create } from 'react-test-renderer';
import { Router } from 'react-router-dom';
import config from '@plone/volto/registry';
import { createMemoryHistory } from 'history';
import ResultItemPreviewImage, {
  previewImageContent,
} from './ResultItemPreviewImage';

const data = {
  '@id': '/testnews1',
  image_field: 'preview_image',
  extras: {
    image_scales: `{"preview_image": [{"filename": "GOPR9639.JPG", "content-type": "image/jpeg", "size": 6446468, "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg", "width": 4000, "height": 3000, "scales": {"icon": {"download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg", "width": 32, "height": 24}, "tile": {"download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg", "width": 64, "height": 48}, "thumb": {"download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg", "width": 128, "height": 96}, "mini": {"download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg", "width": 200, "height": 150}, "preview": {"download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg", "width": 400, "height": 300}, "teaser": {"download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg", "width": 600, "height": 450}, "large": {"download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg", "width": 800, "height": 600}, "larger": {"download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg", "width": 1000, "height": 750}, "great": {"download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg", "width": 1200, "height": 900}, "huge": {"download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg", "width": 1600, "height": 1200}}}]}`,
  },
};

const dataBrokenJson = {
  '@id': '/testnews1',
  image_field: 'preview_image',
  extras: {
    image_scales: `{BROKEN JSON}`,
  },
};

const dataMissingScales = {
  '@id': '/testnews1',
  image_field: 'preview_image',
  extras: {
    image_scales: undefined,
  },
};

describe('ResultItemPreviewImage previewImageContent', () => {
  it('generates image data', () => {
    expect(previewImageContent(data)).toEqual({
      '@id': '/testnews1',
      image_field: 'preview_image',
      image_scales: {
        preview_image: [
          {
            'content-type': 'image/jpeg',
            download:
              '@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg',
            filename: 'GOPR9639.JPG',
            height: 3000,
            scales: {
              great: {
                download:
                  '@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg',
                height: 900,
                width: 1200,
              },
              huge: {
                download:
                  '@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg',
                height: 1200,
                width: 1600,
              },
              icon: {
                download:
                  '@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg',
                height: 24,
                width: 32,
              },
              large: {
                download:
                  '@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg',
                height: 600,
                width: 800,
              },
              larger: {
                download:
                  '@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg',
                height: 750,
                width: 1000,
              },
              mini: {
                download:
                  '@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg',
                height: 150,
                width: 200,
              },
              preview: {
                download:
                  '@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg',
                height: 300,
                width: 400,
              },
              teaser: {
                download:
                  '@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg',
                height: 450,
                width: 600,
              },
              thumb: {
                download:
                  '@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg',
                height: 96,
                width: 128,
              },
              tile: {
                download:
                  '@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg',
                height: 48,
                width: 64,
              },
            },
            size: 6446468,
            width: 4000,
          },
        ],
      },
    });
  });
  describe('fallback', () => {
    it('bad json yields empty image_scales', () => {
      expect(previewImageContent(dataBrokenJson)).toEqual({
        '@id': '/testnews1',
        image_field: 'preview_image',
        image_scales: undefined,
      });
    });
    it('missing scales yields empty image_scales', () => {
      expect(previewImageContent(dataMissingScales)).toEqual({
        '@id': '/testnews1',
        image_field: 'preview_image',
        image_scales: undefined,
      });
    });
  });
});

describe('ResultItemPreviewImage', () => {
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
    history.push = jest.fn();
    window.devicePixelRatio = 1;
  });

  test('normal with default props', () => {
    const component = create(
      <Router history={history}>
        <ResultItemPreviewImage item={data} />
      </Router>,
    );
    const rendered = component.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  test('override via rest', () => {
    const component = create(
      <Router history={history}>
        <ResultItemPreviewImage
          item={data}
          className="foo"
          width="120"
          height="130"
          loading="eager"
          sizes="100vw"
        />
      </Router>,
    );
    const rendered = component.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  describe('legacy image', () => {
    let origComponent;
    let origContentTypeSearchResultAlwaysUseLegacyImage;
    beforeEach(() => {
      origComponent = config.getComponent({ name: 'Image' }).component;
      origContentTypeSearchResultAlwaysUseLegacyImage =
        config.settings.contentTypeSearchResultAlwaysUseLegacyImage;
    });
    afterEach(() => {
      config.getComponent({ name: 'Image' }).component = origComponent;
      config.settings.contentTypeSearchResultAlwaysUseLegacyImage = origContentTypeSearchResultAlwaysUseLegacyImage;
    });

    test('if image component not available on Volto 16', () => {
      config.getComponent({ name: 'Image' }).component = undefined;
      const component = create(
        <Router history={history}>
          <ResultItemPreviewImage
            item={data}
            className="foo"
            width="120"
            height="130"
            loading="eager"
          />
        </Router>,
      );
      const rendered = component.toJSON();
      expect(rendered).toMatchSnapshot();
    });

    test('if contentTypeSearchResultAlwaysUseLegacyImage is true', () => {
      config.settings.contentTypeSearchResultAlwaysUseLegacyImage = true;
      const component = create(
        <Router history={history}>
          <ResultItemPreviewImage
            item={data}
            className="foo"
            width="120"
            height="130"
            loading="eager"
          />
        </Router>,
      );
      const rendered = component.toJSON();
      expect(rendered).toMatchSnapshot();
    });

    test('sizes prop yields error', () => {
      config.settings.contentTypeSearchResultAlwaysUseLegacyImage = true;
      let mockError = jest.spyOn(console, 'error');
      mockError.mockImplementation(() => undefined);
      expect(() =>
        create(
          <Router history={history}>
            <ResultItemPreviewImage
              item={data}
              className="foo"
              width="120"
              height="130"
              loading="eager"
              sizes="100vw"
            />
          </Router>,
        ),
      ).toThrow();
      mockError.mockRestore();
    });
  });
});
