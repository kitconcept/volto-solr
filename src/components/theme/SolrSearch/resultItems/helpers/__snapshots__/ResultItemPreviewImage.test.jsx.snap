// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItemPreviewImage Custom Wrapper 1`] = `
<div
  className="CUSTOM"
>
  <img
    className="previewImage"
    height="125"
    item={
      Object {
        "@id": "/testnews1",
        "image_field": "preview_image",
        "image_scales": Object {
          "preview_image": Array [
            Object {
              "content-type": "image/jpeg",
              "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
              "filename": "GOPR9639.JPG",
              "height": 3000,
              "scales": Object {
                "great": Object {
                  "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                  "height": 900,
                  "width": 1200,
                },
                "huge": Object {
                  "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                  "height": 1200,
                  "width": 1600,
                },
                "icon": Object {
                  "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                  "height": 24,
                  "width": 32,
                },
                "large": Object {
                  "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                  "height": 600,
                  "width": 800,
                },
                "larger": Object {
                  "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                  "height": 750,
                  "width": 1000,
                },
                "mini": Object {
                  "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                  "height": 150,
                  "width": 200,
                },
                "preview": Object {
                  "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                  "height": 300,
                  "width": 400,
                },
                "teaser": Object {
                  "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                  "height": 450,
                  "width": 600,
                },
                "thumb": Object {
                  "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                  "height": 96,
                  "width": 128,
                },
                "tile": Object {
                  "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                  "height": 48,
                  "width": 64,
                },
              },
              "size": 6446468,
              "width": 4000,
            },
          ],
        },
      }
    }
    loading="lazy"
    sizes="250px"
    width="250"
  />
</div>
`;

exports[`ResultItemPreviewImage Wrapper=null removes link 1`] = `
<img
  className="previewImage"
  height="125"
  item={
    Object {
      "@id": "/testnews1",
      "image_field": "preview_image",
      "image_scales": Object {
        "preview_image": Array [
          Object {
            "content-type": "image/jpeg",
            "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
            "filename": "GOPR9639.JPG",
            "height": 3000,
            "scales": Object {
              "great": Object {
                "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                "height": 900,
                "width": 1200,
              },
              "huge": Object {
                "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                "height": 1200,
                "width": 1600,
              },
              "icon": Object {
                "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                "height": 24,
                "width": 32,
              },
              "large": Object {
                "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                "height": 600,
                "width": 800,
              },
              "larger": Object {
                "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                "height": 750,
                "width": 1000,
              },
              "mini": Object {
                "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                "height": 150,
                "width": 200,
              },
              "preview": Object {
                "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                "height": 300,
                "width": 400,
              },
              "teaser": Object {
                "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                "height": 450,
                "width": 600,
              },
              "thumb": Object {
                "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                "height": 96,
                "width": 128,
              },
              "tile": Object {
                "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                "height": 48,
                "width": 64,
              },
            },
            "size": 6446468,
            "width": 4000,
          },
        ],
      },
    }
  }
  loading="lazy"
  sizes="250px"
  width="250"
/>
`;

exports[`ResultItemPreviewImage legacy image Custom Wrapper 1`] = `
<div
  className="CUSTOM"
>
  <img
    className="previewImage"
    height="125"
    item={
      Object {
        "@id": "/testnews1",
        "image_field": "preview_image",
        "image_scales": Object {
          "preview_image": Array [
            Object {
              "content-type": "image/jpeg",
              "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
              "filename": "GOPR9639.JPG",
              "height": 3000,
              "scales": Object {
                "great": Object {
                  "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                  "height": 900,
                  "width": 1200,
                },
                "huge": Object {
                  "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                  "height": 1200,
                  "width": 1600,
                },
                "icon": Object {
                  "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                  "height": 24,
                  "width": 32,
                },
                "large": Object {
                  "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                  "height": 600,
                  "width": 800,
                },
                "larger": Object {
                  "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                  "height": 750,
                  "width": 1000,
                },
                "mini": Object {
                  "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                  "height": 150,
                  "width": 200,
                },
                "preview": Object {
                  "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                  "height": 300,
                  "width": 400,
                },
                "teaser": Object {
                  "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                  "height": 450,
                  "width": 600,
                },
                "thumb": Object {
                  "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                  "height": 96,
                  "width": 128,
                },
                "tile": Object {
                  "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                  "height": 48,
                  "width": 64,
                },
              },
              "size": 6446468,
              "width": 4000,
            },
          ],
        },
      }
    }
    loading="lazy"
    sizes="250px"
    width="250"
  />
</div>
`;

exports[`ResultItemPreviewImage legacy image Wrapper=null removes link 1`] = `
<img
  className="previewImage"
  height="125"
  item={
    Object {
      "@id": "/testnews1",
      "image_field": "preview_image",
      "image_scales": Object {
        "preview_image": Array [
          Object {
            "content-type": "image/jpeg",
            "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
            "filename": "GOPR9639.JPG",
            "height": 3000,
            "scales": Object {
              "great": Object {
                "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                "height": 900,
                "width": 1200,
              },
              "huge": Object {
                "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                "height": 1200,
                "width": 1600,
              },
              "icon": Object {
                "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                "height": 24,
                "width": 32,
              },
              "large": Object {
                "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                "height": 600,
                "width": 800,
              },
              "larger": Object {
                "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                "height": 750,
                "width": 1000,
              },
              "mini": Object {
                "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                "height": 150,
                "width": 200,
              },
              "preview": Object {
                "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                "height": 300,
                "width": 400,
              },
              "teaser": Object {
                "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                "height": 450,
                "width": 600,
              },
              "thumb": Object {
                "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                "height": 96,
                "width": 128,
              },
              "tile": Object {
                "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                "height": 48,
                "width": 64,
              },
            },
            "size": 6446468,
            "width": 4000,
          },
        ],
      },
    }
  }
  loading="lazy"
  sizes="250px"
  width="250"
/>
`;

exports[`ResultItemPreviewImage legacy image if contentTypeSearchResultAlwaysUseLegacyImage is true 1`] = `
<a
  href="/testnews1"
  onClick={[Function]}
>
  <a
    href="/testnews1"
    onClick={[Function]}
  >
    <img
      className="foo"
      height="130"
      loading="eager"
      src="/testnews1/@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg"
      width="120"
    />
  </a>
</a>
`;

exports[`ResultItemPreviewImage legacy image if image component not available on Volto 16 1`] = `
<a
  href="/testnews1"
  onClick={[Function]}
>
  <a
    href="/testnews1"
    onClick={[Function]}
  >
    <img
      className="foo"
      height="130"
      loading="eager"
      src="/testnews1/@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg"
      width="120"
    />
  </a>
</a>
`;

exports[`ResultItemPreviewImage normal with default props 1`] = `
<a
  href="/testnews1"
  onClick={[Function]}
>
  <img
    className="previewImage"
    height="125"
    item={
      Object {
        "@id": "/testnews1",
        "image_field": "preview_image",
        "image_scales": Object {
          "preview_image": Array [
            Object {
              "content-type": "image/jpeg",
              "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
              "filename": "GOPR9639.JPG",
              "height": 3000,
              "scales": Object {
                "great": Object {
                  "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                  "height": 900,
                  "width": 1200,
                },
                "huge": Object {
                  "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                  "height": 1200,
                  "width": 1600,
                },
                "icon": Object {
                  "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                  "height": 24,
                  "width": 32,
                },
                "large": Object {
                  "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                  "height": 600,
                  "width": 800,
                },
                "larger": Object {
                  "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                  "height": 750,
                  "width": 1000,
                },
                "mini": Object {
                  "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                  "height": 150,
                  "width": 200,
                },
                "preview": Object {
                  "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                  "height": 300,
                  "width": 400,
                },
                "teaser": Object {
                  "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                  "height": 450,
                  "width": 600,
                },
                "thumb": Object {
                  "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                  "height": 96,
                  "width": 128,
                },
                "tile": Object {
                  "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                  "height": 48,
                  "width": 64,
                },
              },
              "size": 6446468,
              "width": 4000,
            },
          ],
        },
      }
    }
    loading="lazy"
    sizes="250px"
    width="250"
  />
</a>
`;

exports[`ResultItemPreviewImage override via rest 1`] = `
<a
  href="/testnews1"
  onClick={[Function]}
>
  <img
    className="foo"
    height="130"
    item={
      Object {
        "@id": "/testnews1",
        "image_field": "preview_image",
        "image_scales": Object {
          "preview_image": Array [
            Object {
              "content-type": "image/jpeg",
              "download": "@@images/preview_image-4000-5496e5d01152aa6661428d9e264d3757.jpeg",
              "filename": "GOPR9639.JPG",
              "height": 3000,
              "scales": Object {
                "great": Object {
                  "download": "@@images/preview_image-1200-0723c41835b6498e598780419cc25a4f.jpeg",
                  "height": 900,
                  "width": 1200,
                },
                "huge": Object {
                  "download": "@@images/preview_image-1600-7c8256769b8c48d0392c6a51ef34a139.jpeg",
                  "height": 1200,
                  "width": 1600,
                },
                "icon": Object {
                  "download": "@@images/preview_image-32-e743dcfb64ce39b0660285a04e7f842b.jpeg",
                  "height": 24,
                  "width": 32,
                },
                "large": Object {
                  "download": "@@images/preview_image-800-b30bff7dbaa62fc6aca82639c9f5ee97.jpeg",
                  "height": 600,
                  "width": 800,
                },
                "larger": Object {
                  "download": "@@images/preview_image-1000-e97d87c2a35fdc6262f429262854faed.jpeg",
                  "height": 750,
                  "width": 1000,
                },
                "mini": Object {
                  "download": "@@images/preview_image-200-2a8af40b17e1f5562607a1cf876244b1.jpeg",
                  "height": 150,
                  "width": 200,
                },
                "preview": Object {
                  "download": "@@images/preview_image-400-1a82684a51b55644eef2807c2761b711.jpeg",
                  "height": 300,
                  "width": 400,
                },
                "teaser": Object {
                  "download": "@@images/preview_image-600-71895aac86b72d21dd0f2bad189871c3.jpeg",
                  "height": 450,
                  "width": 600,
                },
                "thumb": Object {
                  "download": "@@images/preview_image-128-88f2831a9f099921f03d3244021b03ab.jpeg",
                  "height": 96,
                  "width": 128,
                },
                "tile": Object {
                  "download": "@@images/preview_image-64-9ad26e8053451aec63586d66e111deb8.jpeg",
                  "height": 48,
                  "width": 64,
                },
              },
              "size": 6446468,
              "width": 4000,
            },
          ],
        },
      }
    }
    loading="eager"
    sizes="100vw"
    width="120"
  />
</a>
`;
